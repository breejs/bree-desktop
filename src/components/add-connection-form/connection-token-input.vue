<script setup>
import { ref } from 'vue';
import useDebounceInput from '@/composables/debounce-input';

const props = defineProps({
  value: {
    type: String,
    default: ''
  }
});

const emit = defineEmits(['update:value']);

const input = ref(null);

const { inputValue, valid, classes } = useDebounceInput(
  props.value,
  '',
  input,
  emit,
  (value) => value === '' || /^[\w-=]+\.[\w-=]+\.?[\w-.+/=]*$/.test(value)
);

defineExpose({ valid });
</script>

<template lang="pug">
.form-floating
  input#connection-token-input.form-control(
    name='token',
    type='text',
    placeholder='',
    v-model='inputValue',
    :class='classes',
    ref='input'
  )
  label(for='connection-token-input')
    | Token
  .invalid-feedback
    | Enter a valid JWT token
</template>
