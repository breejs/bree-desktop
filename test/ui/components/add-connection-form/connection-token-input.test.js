import { storyTests } from '../../_story-tests';

import * as stories from '@/components/add-connection-form/connection-token-input.stories';

storyTests(stories);
