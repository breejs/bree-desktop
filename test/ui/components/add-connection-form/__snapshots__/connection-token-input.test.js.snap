// Vitest Snapshot v1

exports[`story snapshots > Default 1`] = `
"<form novalidate=\\"\\">
  <div class=\\"form-floating\\"><input class=\\"form-control\\" id=\\"connection-token-input\\" name=\\"token\\" type=\\"text\\" placeholder=\\"\\"><label for=\\"connection-token-input\\">Token</label>
    <div class=\\"invalid-feedback\\">Enter a valid JWT token</div>
  </div>
</form>"
`;

exports[`story snapshots > Invalid 1`] = `
"<form novalidate=\\"\\">
  <div class=\\"form-floating\\"><input class=\\"form-control\\" id=\\"connection-token-input\\" name=\\"token\\" type=\\"text\\" placeholder=\\"\\"><label for=\\"connection-token-input\\">Token</label>
    <div class=\\"invalid-feedback\\">Enter a valid JWT token</div>
  </div>
</form>"
`;

exports[`story snapshots > Valid 1`] = `
"<form novalidate=\\"\\">
  <div class=\\"form-floating\\"><input class=\\"form-control\\" id=\\"connection-token-input\\" name=\\"token\\" type=\\"text\\" placeholder=\\"\\"><label for=\\"connection-token-input\\">Token</label>
    <div class=\\"invalid-feedback\\">Enter a valid JWT token</div>
  </div>
</form>"
`;
